primitives/src/script/borrowed.rs

```diff
@@ -49,6 +49,14 @@ internals::transparent_newtype! {
     /// The type is `#[repr(transparent)]` for internal purposes only!
     /// No consumer crate may rely on the representation of the struct!
     ///
+    /// # Hexadecimal strings
+    ///
+    /// Scripts are consensus encoded with a length prefix and as a result of this in some places in
+    /// the eccosystem one will encounter hex strings that include the prefix while in other places
```

Comment:
I think the way it's written is confusing because it's not clear how the seeming inconsistency is a result of consensus encoding. I'd writ something like:

Scripts are slices of bytes however they are encoded differently in various places. The consensus encoding prefixes them with length but encodings that deal with length differently may not add the prefix because they don't need it. E.g. hex-encoded inside json. However you may see hex-encoded consensus encoding as well. Unfortunately one needs to pay attention to the exact specification of whichever encoding is used. To make things clearer we have two variations of encoding and decoding methods with or without the prefix. Note also that `LowerHex` uses the unprefixed version.